### PR TITLE
DFR-1291-amend-application-mandatory-fields

### PR DESCRIPTION
**Pull request by dawudgovuk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFR-1291

### Change description ###
FR Contested - Update Amend Application details event to reflect case creation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json [hunk 1]
```diff
@@ -3442,7 +3442,7 @@
     "CaseEventID": "FR_amendApplication",
     "CaseFieldID": "applicantSolicitorEmail",
     "PageFieldDisplayOrder": 9,
-    "DisplayContext": "MANDATORY",
+    "DisplayContext": "OPTIONAL",
```
Comment (EllisD-B) [inline definitions/contested/json/CaseEventToFields/CaseEventToFields.json:3445]: This field is mandatory during paper case creation

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json [hunk 2]
```diff
@@ -3466,7 +3466,7 @@
     "CaseEventID": "FR_amendApplication",
     "CaseFieldID": "applicantSolicitorConsentForEmails",
     "PageFieldDisplayOrder": 11,
-    "DisplayContext": "MANDATORY",
+    "DisplayContext": "OPTIONAL",
```
Comment (EllisD-B) [inline definitions/contested/json/CaseEventToFields/CaseEventToFields.json:3469]: So is this

**Review by EllisD-B — CHANGES_REQUESTED** [review]

File: definitions/contested/json/CaseEventToFields/CaseEventToFields.json (hunk 1 above)
Comment (dawudgovuk) [reply definitions/contested/json/CaseEventToFields/CaseEventToFields.json:3445]: Ok I've corrected this. It does appear when Applicant is not represented - which didn't look right - could this be a bug?
Should there be this condition like as in FR_newPaperCase?:         "FieldShowCondition": "applicantRepresented=\"Yes\""

Comment (EllisD-B) [reply definitions/contested/json/CaseEventToFields/CaseEventToFields.json]: It doesn't look like there's even an option to select whether the applicant is represented for the amend case event, which would be why there's no field show condition

**Review by rishikrsharma — CHANGES_REQUESTED** [review]
This need to be change - you need to included applicant represented field here. 
At the moment it do not bring Screen if applicant is not represented.

File: definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json [hunk 3]
```diff
@@ -311,7 +311,7 @@
     "ListElementCode": "Organisation.OrganisationID",
     "EventElementLabel": "",
     "FieldDisplayOrder": "",
-    "DisplayContext": "MANDATORY"
+    "DisplayContext": "OPTIONAL"
```
Comment (rishikrsharma) [inline definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json:314]: OrganisationID can't be optional

File: definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json [hunk 4]
```diff
@@ -439,7 +439,7 @@
...
```
Comment (rishikrsharma) [inline definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json:442]: same OrganisationID can't be optional

Comment (dawudgovuk) [reply definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json]: Reverted this as should be Mandatory

File: definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json (hunk 3 above)
Comment (dawudgovuk) [reply definitions/contested/json/CaseEventToComplexTypes/CaseEventToComplexTypes.json]: Reverted this as should be Mandatory

**Review by rishikrsharma — APPROVED** [review]

**Review by EllisD-B — APPROVED** [review]

**Pull request merged**